pref-adjust.lua

```diff
@@ -1,80 +1,125 @@
+    -- Type 5: Plant Likes: "Likes plump helmets for their rounded tops"
+    insert_preference(unit,5,dfhack.matinfo.find("MUSHROOM_HELMET_PLUMP:STRUCTURAL").index)
+--  insert_preference(unit,5,dfhack.matinfo.find("PEACH").index)
 
-    prefcount = #(unit.status.current_soul.preferences)
-    print ("After, unit "..dfhack.TranslateName(dfhack.units.getVisibleName(unit)).." has "..prefcount.." preferences")
+    -- Type 2: Prefers to consume drink: (From plump helmets we get dwarven wine)
+    insert_preference(unit,2,{df.item_type.DRINK,"MUSHROOM_HELMET_PLUMP:DRINK"})
+    -- Type 2: Prefers to consume food: (plump helmets, mushrooms)
+    insert_preference(unit,2,{df.item_type.PLANT,"MUSHROOM_HELMET_PLUMP:MUSHROOM"})
+    -- Type 2: Prefers to consume prepared meals: (quarry bush)
+    insert_preference(unit,2,{df.item_type.FOOD,"BUSH_QUARRY"})
 
-end
+    -- Type 3: Creature detests (TROLL, BIRD_BUZZARD, BIRD_VULTURE, CRUNDLE)
+    insert_preference(unit,3,list_of_creatures.TROLL)
+    insert_preference(unit,3,list_of_creatures.BIRD_BUZZARD)
+    insert_preference(unit,3,list_of_creatures.BIRD_VULTURE)
+    insert_preference(unit,3,list_of_creatures.CRUNDLE)
+end -- end IDEAL profile
+
+if profile == "GOTH" then
+    insert_preference(unit,0,"CREATURE:DWARF:SKIN")
+    insert_preference(unit,4,df.item_type.CORPSE)
+    insert_preference(unit,4,df.item_type.CORPSEPIECE)
+    insert_preference(unit,4,df.item_type.REMAINS)
+    insert_preference(unit,4,df.item_type.COFFIN)
+    insert_preference(unit,7,list_of_colors.BLACK)
+    insert_preference(unit,8,list_of_shapes.CROSS)
+    insert_preference(unit,5,dfhack.matinfo.find("GLUMPRONG").index)
+    insert_preference(unit,2,{df.item_type.DRINK,"WEED_RAT:DRINK"})
+    insert_preference(unit,2,{df.item_type.DRINK,"SLIVER_BARB:DRINK"})
+    insert_preference(unit,2,{df.item_type.PLANT,"TUBER_BLOATED:STRUCTURAL"})
+    insert_preference(unit,3,list_of_creatures.ELF)
+    insert_preference(unit,3,list_of_creatures.HUMAN)
+    insert_preference(unit,3,list_of_creatures.DWARF)
+    if list_of_creatures.DEMON_1 and df.global.world.raws.creatures.all[list_of_creatures.DEMON_1].prefstring[0] then
+        insert_preference(unit,1,list_of_creatures.DEMON_1)
+    end
+    if #df.global.world.poetic_forms.all then
+        insert_preference(unit,9,0) -- this just inserts the first song out of typically many.
+    end
+    if #df.global.world.musical_forms.all then
+        insert_preference(unit,10,0) -- same goes for music
+    end
+    if #df.global.world.dance_forms.all then
+        insert_preference(unit,11,0) -- and dancing
+    end
+end -- end GOTH profile
+
+    prefcount = #(unit.status.current_soul.preferences)
+    print ("After adjusting, unit "..dfhack.TranslateName(dfhack.units.getVisibleName(unit)).." has "..prefcount.." preferences")
```

Comment:
Everywhere you use `TranslateName` (I see 6 places), you should make sure that when you print to the console (like here) you wrap the output in `dfhack.df2console`, like so:
```lua
print(dfhack.df2console(text_with_translated_name))
```
If you're on Windows, this won't make a difference, because the console there is CP437 encoded (same as DF). But on other platforms, any accented/special characters in names will display incorrectly unless you do this.